tests/kernel_performance/hanaperf_postinstall.pm

```diff
@@ -0,0 +1,153 @@
+    assert_script_run("cd /root/sleperf/SLEPerf; ./installer.sh scheduler-service");
+    assert_script_run("cd /root/sleperf/SLEPerf; ./installer.sh common-infra");
+}
+
+sub extract_settings_qaset_config {
+    my $values = shift;
+    my @fields = split(/;/, $values);
+    if (scalar @fields > 0) {
+        foreach my $a_value (@fields) {
+            record_info(${a_value});
+            assert_script_run("echo '${a_value}' >> /root/qaset/config");
+        }
+    }
+}
+
+sub setup_environment {
+    my $qaset_role = get_var('QASET_ROLE', 'HANA');
+    my $mitigation_switch = get_var('MITIGATION_SWITCH', 'mitigations=auto');
+    my $qaset_kernel_tag = get_var('QASET_KERNEL_TAG', '');
+    my $ver_cfg = get_var('VER_CFG');
+    my $disable_services = "qaperf.service chronyd.service firewalld.service";
+
+    # Fill $ver_cfg by default value if it is undefined
+    unless ($ver_cfg) {
+        my $mybuild = check_var('BUILD', 'GM') ? "GM" : "Build" . get_var("BUILD", '');
+        $ver_cfg = "PRODUCT_RELEASE=SLES-" . get_var('VERSION') . ";PRODUCT_BUILD=$mybuild";
+        record_info($ver_cfg);
+    }
+
+    # Disable service
+    assert_script_run("systemctl disable $disable_services");
+
+    # sync time
+    assert_script_run("chronyd -q 'server 0.europe.pool.ntp.org iburst'");
+
+    # set static hostname
+    assert_script_run("hostnamectl hostname `hostname -s`");
+
+    # create basic /root/qaset/config
+    assert_script_run("mkdir -p /root/qaset/");
+    my $qaset_config_file = <<'EOF';
+_QASET_ROLE=HANA
+SQ_TEST_RUN_SET=performance
+SQ_MSG_QUEUE_ENALBE=y
+_QASET_SOFTWARE_TAG=baremetal
+_QASET_SOFTWARE_SUB_TAG=default
+EOF
+    assert_script_run("echo -n '$qaset_config_file' > /root/qaset/config");
+    assert_script_run("echo '_QASET_KERNEL_TAG=$qaset_kernel_tag' >> /root/qaset/config") if $qaset_kernel_tag ne '';
+
+    # The qaset/config need not be updated when role is HANA and ABAP
+    return if (get_var('PROJECT_M_ROLE', "") =~ /PROJECT_M_HANA|PROJECT_M_ABAP/);
+
+    # Extract the openQA parameter: QASET_CONFIG and VER_CFG
+    # Historical reasons that the different settings splitted into two parameters.
+    # VER_CFG="PRODUCT_RELEASE=SLES-15-SP3;PRODUCT_BUILD=202109"
+    extract_settings_qaset_config(get_var("QASET_CONFIG", ''));
+    extract_settings_qaset_config($ver_cfg);
+}
+
+sub os_update {
+    my $update_repo_url = shift;
+    my $zypper_repo_path = "/etc/zypp/repos.d";
+
+    assert_script_run("wget -N -P $zypper_repo_path $update_repo_url 2>&1");
+    zypper_call("--gpg-auto-import-keys ref");
+    zypper_call("dup", timeout => 1800);
+}
+
+sub handle_repo_and_package {
+    my $package_install = "qa_lib_ctcs2 wget bc bzip2 screen cpupower pciutils lsscsi smartmontools netcat-openbsd libltdl7 unzip lvm hana_insserv_compat";
+    my $package_remove = "snapper-zypp-plugin";
+
+    # Add QA:HEAD repo
+    add_qa_head_repo;
+
+    # Install mandatory packages for SLE16
+    zypper_call("install $package_install");
```

Comment:
Is `$package_install` needed. Why not to just have the list of packages directly here?

Reply:
Agree with Michele. I think I would go for something like:
```
sub handle_repo_and_package {
    # Add QA:HEAD repo
    add_qa_head_repo;
    # Install mandatory packages for SLE16
    zypper_call('in qa_lib_ctcs2 wget bc bzip2 screen cpupower pciutils lsscsi smartmontools netcat-openbsd libltdl7 unzip lvm hana_insserv_compat');
    zypper_call('rm snapper-zypp-plugin');
}
```